pandas/errors/__init__.py

```diff
@@ -456,12 +456,30 @@ class AttributeConflictWarning(Warning):
     """
 
 
+class DatabaseError(OSError):
+    """
+    Error is raised when executing sql with bad syntax or sql that throws an error.
+
+    Examples
+    --------
+    >>> import pandas.io.sql as sql
```

Comment:
Can just call `pd.read_sql` below

Reply:
I don't think it's necessary to demo `sql.execute` since it's not public

Reply:
Updated. Maybe an obvious question, how do you know if something is public or not?

Reply:
Good question (and definitely not obvious). Anything in the pandas namespace `import pandas as pd` or `pandas/api` is public. Users can access other namespaces but they are not "public" e.g. `pandas.core`

Reply:
Gotcha, thank you for the information!